grapher/scan.py

```diff
@@ -132,11 +132,22 @@ def pkgToUnit(pkg: Dict) -> List[Unit]:
     if len(test_files) == 0:
         return [unit]
 
+    # Collect files as candidates of modules.
+    modules = []
```

Comment:
Ah, I see. What's a real-world example? Let's include a test case that covers this scenario.

Factor this out into a separate function with a docstring that includes the explanation you just gave above.